src/herder/TxQueueLimiter.cpp

```diff
@@ -117,22 +120,29 @@ TxQueueLimiter::canAddTx(TransactionFrameBasePtr& newTx,
     {
         if (feeRate3WayCompare(*it, newTx) >= 0)
         {
-            return false;
+            int64 minFee;
+            // compute a fee strictly better than `it`
+            // this won't overflow because fee bids for valid transactions are
+            // too small even when scaled 101x
+            minFee = bigDivide((*it)->getFeeBid(), newTx->getNumOperations(),
+                               (*it)->getNumOperations(), Rounding::ROUND_UP) +
+                     1;
+            return std::make_pair(false, minFee);
         }
         // ensure that this transaction is not from the same account
         auto& tx = *it;
         if (tx->getSourceID() == id)
         {
-            return false;
+            return std::make_pair(false, 0ll);
         }
         auto curOps = tx->getNumOperations();
         if (neededOps <= curOps)
         {
-            return true;
+            return std::make_pair(true, 0ll);
         }
         neededOps -= curOps;
     }
-    return false;
+    return std::make_pair(false, 0ll);
```

Comment:
This case is interesting... it means that the transaction just didn't fit in the queue even if you remove everything else? Or am I missing something? If I'm reading this correctly, then we should probably add a comment here that explains what this means. And we should probably return a special value which means something like "submit to another node".

Reply:
added a comment to clarify

Reply:
I don't think we should add a new return value: it's just a variation on "try again later" type of situation for the client

Reply:
Well, it's a bit different. If you get here, you will _never_ succeed no matter how long you wait unless the node changes configuration. Basically the way I see it is, if you get here then you are totally broken (eg. send an alert to operations) whereas if you hit any of the other cases then you can just back off and wait.

Reply:
OK for operators, this makes sense. I am adding a warning when the limiter is configured that way.